src/algebra/associated.lean

```diff
@@ -627,6 +627,34 @@ calc is_unit (associates.mk a) ↔ a ~ᵤ 1 :
     by rw [is_unit_iff_eq_one, one_eq_mk_one, mk_eq_mk_iff_associated]
   ... ↔ is_unit a : associated_one_iff_is_unit
 
+section unique_units
+
+variable [unique (αˣ)]
+
+/-- `associates.mk` as a monoid equiv -/
+noncomputable def mk_monoid_equiv : α ≃* associates α :=
+mul_equiv.of_bijective
+  (@associates.mk_monoid_hom α _) ⟨associates.mk_injective, associates.mk_surjective⟩
+
+@[simp]
+lemma mk_monoid_equiv_apply {a : α} :
```

Comment:
Why `a` is implicit in all these lemmas? It's probably not so important, but it's not mentioned in any assumption, right?

Reply:
this already exists, though, as `associates_equiv_of_unique_units`.